image-inspector.go

```diff
@@ -126,8 +165,14 @@ func main() {
 	if _, err := client.InspectImage(*image); err != nil {
 		log.Printf("Pulling image %s", *image)
 		imagePullOption := docker.PullImageOptions{Repository: *image}
-		imagePullAuth := docker.AuthConfiguration{} // TODO: support authentication
-		if err := client.PullImage(imagePullOption, imagePullAuth); err != nil {
+		imagePullAuths := getAuthConfigs(dockercfg, username, password_file)
+		// Try all the possible auth's from the config file
+		for _, auth := range imagePullAuths.Configs {
+			if err = client.PullImage(imagePullOption, auth); err == nil {
+				break
```

Comment:
@enoodle you can just `log.Fatalf` here (and drop the need for the `err` variable up above).